lib/CL/pocl_tracing.c

```diff
@@ -0,0 +1,299 @@
+#include <stdio.h>
+#include <string.h>
+#include "pocl_tracing.h"
+#include "pocl_runtime_config.h"
+
+#ifdef LTTNG_UST_AVAILABLE
+#include "pocl_lttng.h"
+#endif
+
+static int tracing_initialized = 0;
+static uint8_t event_trace_filter = 0xF;
+
+static const char *status_to_str[] = {
+  "complete",
+  "running",
+  "submitted",
+  "queued"
+};
+
+static const struct pocl_event_tracer text_logger;
+static const struct pocl_event_tracer lttng_tracer;
+
+/**
+ * List of tracers
+ */
+static const struct pocl_event_tracer *pocl_event_tracers[] = {
+  &text_logger,
+#ifdef LTTNG_UST_AVAILABLE
+  &lttng_tracer,
+#endif
+};
+
+#define POCL_TRACER_COUNT (sizeof(pocl_event_tracers) / sizeof((pocl_event_tracers)[0]))
+
+static const struct pocl_event_tracer *event_tracer = NULL;
+
+int pocl_event_updated(cl_event event, int status)
```

Comment:
Please check: http://www.gnu.org/prep/standards/html_node/Formatting.html#Formatting -- should have the return value in its own line etc.